Translate tasks/inject-data-application/define-environment-variable-container in Korean (#15606)
kubernetes/website: content/ko/docs/tasks/inject-data-application/define-environment-variable-container.md

---
title: 컨테이너를 위한 환경 변수 정의하기
content_template: templates/task
weight: 20
---

{{% capture overview %}}

> 본 페이지는 어떻게 쿠버네티스 파드 안에서 컨테이너를 위한 환경 변수를

Reviewer: [원문]
This page shows how to define environment variables for a container
in a Kubernetes Pod.
파드 안에서 컨테이너의 환경 변수를 정의한다기 보다는,
파드에 포함되어 있는 컨테이너의 환경 변수를 정의한다는 의미인 것 같습니다.
```suggestion
본 페이지는 쿠버네티스 파드의 컨테이너를 위한 환경 변수를
```

정의하는 방법에 대해 설명한다.

{{% /capture %}}

{{% capture prerequisites %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

{{% /capture %}}

{{% capture steps %}}

# 컨테이너를 위한 환경 변수 정의하기

> 파드를 생성할 때, 파드 안에서 실행하는 컨테이너를 위한 환경 변수를 설정할 

Reviewer: [원문]
When you create a Pod, you can set environment variables for the containers
that run in the Pod.
원문을 고려하여 조금 더 자연스럽게 수정해보았습니다. 참고하시기 바랍니다.
```suggestion
파드를 생성할 때, 파드 안에서 동작하는 컨테이너를 위한 환경 변수를 설정할 
```

> 수 있다. 환경변수를 설정하려면, 구성 파일에 `env`나 `envFrom` 필드를

Reviewer: 일관성 유지 차원에서 제안드립니다.
```suggestion
수 있다. 환경 변수를 설정하려면, 구성 파일에 `env`나 `envFrom` 필드를
```

> 포함시켜야한다.

Reviewer: ```suggestion
포함시켜야 한다.
```

이 예제에서, 한 개의 컨테이너를 실행하는 파드를 생성한다. 파드를 위한 구성
파일은 `DEMO_GREETING` 이라는 이름과 `"Hello from the environment"`이라는

> 값을 가지는 환경 변수를 정의한다. 다음은 파드를 위한 구성 파일 예시이다.

Reviewer: 코드의 유지보수를 위해서 원문과 라인 번호를 동일하게 유지하고, 가능한 한 라인의 가로폭을 원문과 유사하게 유지하는 것을 가이드하고 있습니다. 참고: [한글화 스타일 가이드](https://kubernetes.io/ko/docs/contribute/localization_ko/#%EA%B0%80%EB%A1%9C%ED%8F%AD%EC%9D%80-%EC%9B%90%EB%AC%B8%EC%9D%84-%EB%94%B0%EB%A6%84)

[원문]
라인 32: value `"Hello from the environment"`. Here is the configuration file for the
라인 33: Pod:

```suggestion
값을 가지는 환경 변수를 정의한다. 다음은 파드를 위한 구성 파일 
예시이다.
```

{{< codenew file="pods/inject/envars.yaml" >}}

1. YAML 구성 파일을 활용해 파드를 생성한다.

   ```shell
   kubectl apply -f https://k8s.io/examples/pods/inject/envars.yaml
   ```

> 2. 실행 중인 파드들의 목록을 조회한다.

Author: ![image](https://user-images.githubusercontent.com/5652281/62335398-a984e180-b506-11e9-8600-a82124af0889.png)
개행이 일부 잘못되어서 그런지 항목 숫자가 1에서 늘어나지 않았네요. 이 또한 다음 커밋에서 해결하겠습니다.

> 2. 실행 중인 파드들의 목록을 조회한다.

Reviewer: ```suggestion
1. 실행 중인 파드들의 목록을 조회한다.
```
Hugo에서 자동적으로 리스트 번호를 컴파일합니다. 원문과 동일하게 유지하시면 되겠습니다.

   ```shell
   kubectl get pods -l purpose=demonstrate-envars
   ```

>    출력은 아래의 것과 비슷할 것이다.

Author: 4번째 항목과의 일관성을 유지하기 위해 아래와 같이 변경하겠습니다.
`출력은 아래와 비슷할 것이다.`
https://github.com/kubernetes/website/pull/15606/files/3ece8f771f9e5ec8aba2bf38bf273234d02e4378#r310376794

   ```
   NAME            READY     STATUS    RESTARTS   AGE
   envar-demo      1/1       Running   0          9s
   ```

> 3. 파드 안에 실행되고 있는 컨테이너의 셸에 접근한다.

Reviewer: ```suggestion
1. 파드 안에 실행되고 있는 컨테이너의 셸에 접근한다.
```

   ```shell
   kubectl exec -it envar-demo -- /bin/bash
   ```

> 4. 셸 안에서, 환경 변수를 나열하기 위해 `printenv` 명령을 실행한다.

Reviewer: ```suggestion
1. 셸 안에서, 환경 변수를 나열하기 위해 `printenv` 명령을 실행한다.
```

> 4. 셸 안에서, 환경 변수를 나열하기 위해 `printenv` 명령을 실행한다.

Reviewer: ```suggestion
4. 셸 안에서, 환경 변수를 나열하기 위해 `printenv` 커맨드을 실행한다.
```
[한글화 용어집](https://docs.google.com/spreadsheets/d/1BWKq_HYmt_p1RKzQJoPfCb4ZgBgxH_q3_mf9ujCmniw/edit#gid=0)에서는 아래와 같이 번역어를 정의하고 있습니다.

Command | 커맨드
-- | --

Author: 단어를 바꾸면서 자연스럽게 하기 위해 `커맨드를`로 수정하겠습니다!

   ```shell
   root@envar-demo:/# printenv
   ```

>    출력은 아래의 것과 비슷할 것이다.

Reviewer: 조금 자연스럽게 변경해 보았습니다. 참고하시기 바랍니다.
```suggestion
   출력은 아래와 비슷할 것이다.
```

   ```
   NODE_VERSION=4.4.2
   EXAMPLE_SERVICE_PORT_8080_TCP_ADDR=10.3.245.237
   HOSTNAME=envar-demo
   ...
   DEMO_GREETING=Hello from the environment
   DEMO_FAREWELL=Such a sweet sorrow
   ```

> 5. 셸에서 빠져나오기 위해, `exit`을 입력한다.

Reviewer: ```suggestion
1. 셸에서 빠져나오기 위해, `exit`을 입력한다.
```

{{< note >}}

> `env` 나 `envFrom` 필드를 이용해 설정된 환경변수들은 컨테이너 이미지 

Reviewer: ```suggestion
`env` 나 `envFrom` 필드를 이용해 설정된 환경 변수들은 컨테이너 이미지 
```

안에서 명시된 어떠한 환경 변수들보다 더 우선시된다.
{{< /note >}}

## 설정 안에서 환경 변수 사용하기

> 파드의 구성 파일 안에서 정의한 환경 변수는 파드의 컨테이너를 위해 설정하는 명령들과 인자들과 같이, 구성 파일 안의 다른 곳에서 사용할 수 있다. 아래의 구성 파일 예시에서, `GREETING`, `HONORIFIC`, 그리고 `NAME` 환경 변수들이 각각 `Warm greetings to`, `The Most honorable`, 그리고 `Kubernetes`로 설정되어 있다. 이들 환경변수들은 이후 `env-print-demo` 컨테이너에 전달되어 CLI 인자에서 사용된다.

Reviewer: ```suggestion
파드의 구성 파일 안에서 정의한 환경 변수는 파드의 컨테이너를 위해 설정하는 커맨드들과 인자들과 같이, 구성 파일 안의 다른 곳에서 사용할 수 있다. 아래의 구성 파일 예시에서, `GREETING`, `HONORIFIC`, 그리고 `NAME` 환경 변수들이 각각 `Warm greetings to`, `The Most honorable`, 그리고 `Kubernetes`로 설정되어 있다. 이들 환경 변수들은 이후 `env-print-demo` 컨테이너에 전달되어 CLI 인자에서 사용된다.
```

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: print-greeting
spec:
  containers:
  - name: env-print-demo
    image: bash
    env:
    - name: GREETING
      value: "Warm greetings to"
    - name: HONORIFIC
      value: "The Most Honorable"
    - name: NAME
      value: "Kubernetes"
    command: ["echo"]
    args: ["$(GREETING) $(HONORIFIC) $(NAME)"]
```

> 컨테이너가 생성되면, `echo Warm greetings to The Most Honorable Kubernetes` 명령이 컨테이너에서 실행된다.

Reviewer: ```suggestion
컨테이너가 생성되면, `echo Warm greetings to The Most Honorable Kubernetes` 커맨드가 컨테이너에서 실행된다.
```

{{% /capture %}}

{{% capture whatsnext %}}

> *  [환경 변수](/docs/tasks/inject-data-application/environment-variable-expose-pod-information/)에 대해 알아본다.

Reviewer: ```suggestion
* [환경 변수](/docs/tasks/inject-data-application/environment-variable-expose-pod-information/)에 대해 알아본다.
```

> * [시크릿을 환경변수로 사용하기](/docs/user-guide/secrets/#using-secrets-as-environment-variables)에 대해 알아본다.

Reviewer: ```suggestion
* [시크릿을 환경 변수로 사용하기](/docs/user-guide/secrets/#using-secrets-as-environment-variables)에 대해 알아본다.
```
